openwisp_monitoring/monitoring/tests/test_monitoring_notifications.py

```diff
@@ -148,18 +156,24 @@ def test_general_check_threshold_crossed_for_long_time(self):
 
         with self.subTest('Test no recovery notification yet (tolerance not passed)'):
             m.write(50)
+            m.refresh_from_db()
             self.assertFalse(m.is_healthy)
             self.assertEqual(Notification.objects.count(), 1)
 
         with self.subTest('Tolerance still not passed, not expecting a recovery yet'):
             with freeze_time(start_time + timedelta(minutes=2)):
                 m.write(51)
+            m.refresh_from_db()
             self.assertFalse(m.is_healthy)
             self.assertEqual(Notification.objects.count(), 1)
 
         with self.subTest('Test recovery notification after tolerance is passed'):
             with freeze_time(ten_minutes_after):
                 m.write(50)
+            # TODO: This sleep needs to be removed, currently it has been added
+            # to avoid this test being flaky due to async task.
+            sleep(1)
```

Comment:
during tests celery tasks are executed synchronously: https://github.com/openwisp/openwisp-monitoring/blob/3d55db0a94933515ce410989c9e8c2ca24b61088/tests/openwisp2/settings.py#L165-L170

Therefore this sleep should not be needed.